jsFiles/app.js

```diff
@@ -60,6 +60,13 @@ function fetchData(apiUrl, searchCity = false) {
     // of an asynchronous operation and its resulting value.
     .then((data) => {
       console.log(data);
+      // this code reports that city wasn't found
+      // because with code 404, it doesn't go directly to catch
+      // and it breakes with not finding temp in data.main.
+      if (data.cod == 404) {
```

Comment:
FYI you can also check the status code before parsing the response object. This is the standard browser API to detect the HTTP response status code

https://developer.mozilla.org/en-US/docs/Web/API/Response/status

Reply:
It is now using response.status